src/Language/PureScript/Docs/Prim.hs

```diff
@@ -377,70 +377,59 @@ coercible :: Declaration
+  , "`Coercible a a` holds."
   , ""
-  , "    instance coercibleReflexive :: Coercible a a"
+  , "* _symmetry_, if a type `a` can be coerced to some other type `b`, then `b`"
+  , "can also be coerced back to `a`: `Coercible a b` implies `Coercible b a`."
   , ""
-  , "Second, Coercible obeys symmetry - if a type `a` can be coerced to some"
-  , "other type `b`, then `b` can also be coerced back to `a`:"
+  , "* _transitivity_, if a type `a` can be coerced to some other type `b` which"
+  , "can be coerced to some other type `c`, then `a` can also be coerced to `c`:"
+  , "`Coercible a b` and `Coercible b c` imply `Coercible a c`."
   , ""
-  , "    instance coercibleSymmetric :: Coercible a b => Coercible b a"
+  , "* Newtypes can be freely wrapped and unwrapped when their constructor is"
+  , "in scope:"
   , ""
-  , "Third, Coercible obeys transitivity - if a type `a` can be coerced to some"
-  , "other type `b` which can be coerced to some other type `c`, then `a` can"
-  , "also be coerced to `c`:"
+  , "      newtype Age = Age Int"
   , ""
-  , "    instance coercibleTransitive :: (Coercible a b, Coercible b c) => Coercible a c"
+  , "`Coercible Int Age` and `Coercible Age Int` hold since `Age` has the same"
+  , "runtime representation than `Int`."
   , ""
-  , "Fourth, for every type constructor there is an instance that allows one"
-  , "to coerce under the type constructor (`data` or `newtype`). For example,"
-  , "given a definition:"
+  , "Newtype constructors have to be in scope to preserve abstraction. It's"
+  , "common to declare a newtype to encode some invariants (non emptiness of"
+  , "arrays with `Data.Array.NonEmpty.NonEmptyArray` for example), hide its"
+  , "constructor and export smart constructors instead. Without this restriction,"
+  , "the guarantees provided by such newtypes would be void."
   , ""
-  , "data D a b = D a"
+  , "* Terms of different types are not coercible:"
   , ""
-  , "there is an instance:"
+  , "      data Maybe a = Nothing | Just a"
+  , "      data Either a b = Left a | Right b"
   , ""
-  , "    instance coercibleConstructor :: Coercible a a' => Coercible (D a b) (D a' b')"
+  , "`Coercible (Maybe b) (Either a b)` does not hold. Those types don't share a"
+  , "common runtime representation so coercing between them would be unsafe."
   , ""
-  , "Note that, since the type variable `a` plays a role in `D`'s representation,"
-  , "we require that the types `a` and `a'` are themselves `Coercible`. However,"
-  , "since the variable `b` does not play a part in `D`'s representation (a type"
-  , "such as `b` is thus typically referred to as a \"phantom\" type), `b` and `b'`"
-  , "can differ arbitrarily."
+  , "* Terms of the same types (`data` or `newtype`) may be coercible, depending"
+  , "on the _roles_ of their parameters. Roles are documented in [the PureScript"
+  , "language reference](https://github.com/purescript/documentation/blob/master/language/Roles.md)."
```

Comment:
I think the ‘terms of different/the same types’ terminology is a little confusing. How about:

* Two types of kind `Type` may be coercible if the following two conditions are met:
  * Their heads must be the same. For example, `Maybe a` and `Either a b` don't satisfy this condition because `Maybe` and `Either` are different. Those types don't share a common runtime representation so coercing between them would be unsafe.
  * Their arguments may need to be identical or coercible depending on the _roles_ of the head's type parameters. Roles are documented in [the PureScript language reference](https://github.com/purescript/documentation/blob/master/language/Roles.md).

Reply:
Oh right, I see how this can be confusing.

I didn’t manage to render a nested list in the generated documentation, so I slightly adapted your suggestion. Let me know what you think!

Reply:
I think this meaning of "head" is actually not quite correct. This is something I've only recently become aware of, so it's possible you even picked it up from me from something I've written elsewhere! But I now suspect it's based on a common misunderstanding of "instance head", in which the "head" of the instance
```
instance semigroupEffect :: Semigroup a => Semigroup (Effect a)
```
is understood to be `Effect` because it's the first type that appears; in fact the head of that instance is `Effect a`. See e.g. https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/exts/instances.html#instance-declarations-and-resolution

Reply:
Huh! TIL, I guess. What *would* you use as a succinct way to describe, um, the leftmost type constructor in a type after any infix type constructors are normalized to prefix?

Reply:
I don't have a good answer for that, unfortunately. "The type being applied in a type application"?

Reply:
At a quick glance, the [coercible paper](https://www.microsoft.com/en-us/research/uploads/prod/2018/05/coercible-JFP.pdf) uses ‘head’ in a few places to mean this I think. So maybe we're okay after all?

Reply:
Ah, good point. I suppose if the authors of that paper are using it that way then it's hard to argue that it's incorrect. That is unfortunate though; it's awkward to have to clarify whether you're talking about a whole instance head or just an, uh, instance head head.

Reply:
Yeah I’m fine with “head” here. I believe the original usage of the term is simply the leftmost applied thing in a term. It’s where the phrase “weak head normal form” comes from.

Maybe the instance head is generalized from this, once we got multiparam typeclasses? idk

Reply:
Here’s a reference for this usage of head position: https://arxiv.org/pdf/1701.04918.pdf